Add vscode workaround into troubleshoorting (#27800)
Azure/azure-sdk-for-net: sdk/identity/Azure.Identity/TROUBLESHOOTING.md

|No Managed Identity endpoint found|The application attempted to authenticate before an identity was assigned to its pod|Verify the pod is labeled correctly. This also occurs when a correctly labeled pod authenticates before the identity is ready. To prevent initialization races, configure NMI to set the Retry-After header in its responses (see [Pod Identity documentation](https://azure.github.io/aad-pod-identity/docs/configure/feature_flags/#set-retry-after-header-in-nmi-response)).

## Troubleshoot `VisualStudioCodeCredential` Authentication Issues

> It is a known issue that `VisualStudioCodeCredential` does not work well with Azure Account extension newer than **version 0.9.11**. If you are using latest version of Azure Account extension, please downgrade to **version 0.9.11** to make `VisualStudioCodeCredential` work correctly.

Reviewer: Is it possible to include any specific error(s) so that people searching the internet for the error message will find this text more easily?

Author: Unfortunately, depending on different situations, user may see different errors. (at least we received two different errors from Python users).

> It is a known issue that `VisualStudioCodeCredential` does not work well with Azure Account extension newer than **version 0.9.11**. If you are using latest version of Azure Account extension, please downgrade to **version 0.9.11** to make `VisualStudioCodeCredential` work correctly.

Reviewer 2: - Should we say "does not work" instead of "does not work well"?
- Link the 1st occurrence of the "Azure Account extension" text to https://marketplace.visualstudio.com/items?itemName=ms-vscode.azure-account.
- Instead of "using latest version", we should say "0.10.0 or later".

> > It is a [known issue](https://github.com/Azure/azure-sdk-for-net/issues/27263) that `VisualStudioCodeCredential` does not work with [Azure Account extension](https://marketplace.visualstudio.com/items?itemName=ms-vscode.azure-account) newer than **version 0.9.11**. If you are using Azure Account extension version 0.10.0 or later, downgrading to **version 0.9.11** will resolve this issue. A long term fix to this problem is in progress.

Reviewer 2: ```suggestion
> It's a [known issue](https://github.com/Azure/azure-sdk-for-net/issues/27263) that `VisualStudioCodeCredential` doesn't work with [Azure Account extension](https://marketplace.visualstudio.com/items?itemName=ms-vscode.azure-account) versions newer than **0.9.11**. If you're using Azure Account extension version 0.10.0 or later, downgrading to **version 0.9.11** will resolve this issue. A long-term fix to this problem is in progress.
```
